drivers/wifi/simplelink_support.c

```diff
@@ -0,0 +1,616 @@
+				    "BSSID: %x:%x:%x:%x:%x:%x on error: %d",
+				    event_data->SsidName, event_data->Bssid[0],
+				    event_data->Bssid[1], event_data->Bssid[2],
+				    event_data->Bssid[3], event_data->Bssid[4],
+				    event_data->Bssid[5],
+				    event_data->ReasonCode);
+			sl_conn.error = event_data->ReasonCode;
+		}
+		memset(&(sl_conn.ssid), 0x0, sizeof(sl_conn.ssid));
+		memset(&(sl_conn.bssid), 0x0, sizeof(sl_conn.bssid));
+
+		/* Continue the notification callback chain... */
+		nwp.cb(NET_EVENT_WIFI_CMD_DISCONNECT_RESULT, &sl_conn);
+		break;
+
+	case SL_WLAN_EVENT_STA_ADDED:
+		memcpy(&(sl_conn.bssid), wlan_event->Data.STAAdded.Mac,
+		       SL_WLAN_BSSID_LENGTH);
+		SYS_LOG_INF("\n[WLAN EVENT] STA was added to AP: "
+			    "BSSID: %x:%x:%x:%x:%x:%x",
+			    sl_conn.bssid[0], sl_conn.bssid[1],
+			    sl_conn.bssid[2], sl_conn.bssid[3],
+			    sl_conn.bssid[4], sl_conn.bssid[5]);
+		break;
+	case SL_WLAN_EVENT_STA_REMOVED:
+		memcpy(&(sl_conn.bssid), wlan_event->Data.STAAdded.Mac,
+		       SL_WLAN_BSSID_LENGTH);
+		SYS_LOG_INF("\n[WLAN EVENT] STA was removed from AP: "
+			    "BSSID: %x:%x:%x:%x:%x:%x",
+			    sl_conn.bssid[0], sl_conn.bssid[1],
+			    sl_conn.bssid[2], sl_conn.bssid[3],
+			    sl_conn.bssid[4], sl_conn.bssid[5]);
+
+		memset(&(sl_conn.bssid), 0x0, sizeof(sl_conn.bssid));
+		break;
+	default:
+		SYS_LOG_ERR("\n[WLAN EVENT] Unexpected event [0x%lx]",
+			    wlan_event->Id);
+		break;
+	}
+}
+
+/**
+ *  @brief SimpleLinkNetAppEventHandler
+ *
+ *  This handler gets called whenever a Netapp event is reported
+ *  by the host driver / NWP.
+ *
+ *  @note See the CC3120/CC3220 NWP programmer's guide (SWRU455)
+ *	  section 5.7.
+ */
+void SimpleLinkNetAppEventHandler(SlNetAppEvent_t *netapp_event)
+{
+	u32_t i = 0;
+
+	SlIpV4AcquiredAsync_t *event_data = NULL;
+
+	if (!netapp_event) {
+		return;
+	}
+
+	switch (netapp_event->Id) {
+	case SL_DEVICE_EVENT_DROPPED_NETAPP_IPACQUIRED:
+		SET_STATUS_BIT(nwp.status, STATUS_BIT_IP_ACQUIRED);
+
+		/* Ip Acquired Event Data */
+		event_data = &netapp_event->Data.IpAcquiredV4;
+		sl_conn.ip_addr = event_data->Ip;
+
+		/* Gateway IP address */
+		sl_conn.gateway_ip = event_data->Gateway;
+
+		SYS_LOG_INF("\n[NETAPP EVENT] IP set to: IPv4=%d.%d.%d.%d, "
+		       "Gateway=%d.%d.%d.%d",
+		       SL_IPV4_BYTE(sl_conn.ip_addr, 3),
+		       SL_IPV4_BYTE(sl_conn.ip_addr, 2),
+		       SL_IPV4_BYTE(sl_conn.ip_addr, 1),
+		       SL_IPV4_BYTE(sl_conn.ip_addr, 0),
+
+		       SL_IPV4_BYTE(sl_conn.gateway_ip, 3),
+		       SL_IPV4_BYTE(sl_conn.gateway_ip, 2),
+		       SL_IPV4_BYTE(sl_conn.gateway_ip, 1),
+		       SL_IPV4_BYTE(sl_conn.gateway_ip, 0));
+
+		nwp.cb(NET_EVENT_IPV4_ADDR_ADD, &sl_conn);
+		break;
+
+	case SL_DEVICE_EVENT_DROPPED_NETAPP_IPACQUIRED_V6:
+		SET_STATUS_BIT(nwp.status, STATUS_BIT_IPV6_ACQUIRED);
+
+		for (i = 0; i < 4; i++) {
+			sl_conn.ipv6_addr[i] =
+			  netapp_event->Data.IpAcquiredV6.Ip[i];
+		}
+
+		SYS_LOG_INF("[NETAPP EVENT] IP Acquired: IPv6=");
```

Comment:
In practice splitting syslog prints will not work very well, the output will look very weird because each syslog call will print the header.  You could just call net_addr_ntop() to get the string presentation of IPv6 address and then print everything in one go.

So perhaps something like
```
#if SYS_LOG_LEVEL >= 3
{
    char ipv6_addr[NET_IPV6_ADDR_LEN];

    net_addr_ntop(AF_INET6, sl_conn.ipv6_addr, ipv6_addr, sizeof(ipv6_addr));
    SYS_LOG_INF(".....");
}
#endif
```

Reply:
Noticed error in my comment, it should have been "#if SYS_LOG_LEVEL >= SYS_LOG_LEVEL_INFO" instead (not good idea to use hardcoded numbers here).

Reply:
Good suggestion, I missed that porting an older version of the handler from printf's.